src/test/app/doctest.cpp

```diff
@@ -1,2 +1,14 @@
 #define DOCTEST_CONFIG_IMPLEMENT_WITH_MAIN
+
+// FIXME: DOCTEST doesn't support baremetal targets without mutex & pthreads
+#ifndef __linux
+#define DOCTEST_CONFIG_NO_POSIX_SIGNALS
+namespace std {
```

Comment:
Hack: doctest doesn't support BM, and mutex is missing - workaround to hide compile doctest errors (to check if other issues appear).